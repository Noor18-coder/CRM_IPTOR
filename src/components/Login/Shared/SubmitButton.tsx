export interface Props {
  title: string;
  isSubmit: boolean;
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

export const SubmitButton: React.FC<Props> = ({ title, isSubmit, onClick }) => {
  return (
    <button type="submit" className={isSubmit ? 'iptor-login-btn' : 'iptor-login-btn disable-button'} onClick={onClick}>
      {title}
    </button>
  );
};
