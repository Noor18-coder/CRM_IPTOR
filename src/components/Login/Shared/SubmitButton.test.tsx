import renderer from 'react-test-renderer';
import { SubmitButton } from './SubmitButton';

const onClick = jest.fn();

describe('[Login Shared] SubmitButton', () => {
  it('should renders correctly with isSubmit false', () => {
    const tree = renderer.create(<SubmitButton title="The Button" isSubmit={false} onClick={onClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders correctly with isSubmit true', () => {
    const tree = renderer.create(<SubmitButton title="The Button" isSubmit onClick={onClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
