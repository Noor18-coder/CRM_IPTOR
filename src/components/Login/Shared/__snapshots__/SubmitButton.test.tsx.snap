// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Login Shared] SubmitButton should renders correctly with isSubmit false 1`] = `
<button
  className="iptor-login-btn disable-button"
  onClick={[MockFunction]}
  type="submit"
>
  The Button
</button>
`;

exports[`[Login Shared] SubmitButton should renders correctly with isSubmit true 1`] = `
<button
  className="iptor-login-btn"
  onClick={[MockFunction]}
  type="submit"
>
  The Button
</button>
`;
